src/eventer/eventer_aco_opset.c

```diff
@@ -193,19 +231,34 @@ eventer_aco_simple_asynch_wrapper(eventer_t e, int mask, void *closure, struct t
     simple_ctx->func(simple_ctx->closure);
   }
   if(mask == EVENTER_ASYNCH) {
+    bool zero;
     aco_t *co = simple_ctx->co;
+    ck_pr_dec_32_zero(&simple_ctx->gate->ref_cnt, &zero);
     free(simple_ctx);
```

Comment:
same.  free(simple_ctx) but not gate, i think since this is the ASYNCH path it just needs a comment